### ISSUE TITLE
[Suggestion] Archiving a tournament

### ISSUE THREAD

**Issue by franga2000** [TabbycatDebate/tabbycat#1928]
Hi again! This is another one of those suggestions that I intend to implement for our own use and want to ask for feedback here so it can hopefully be merged. *And hopefully I actually do it this time*

**The problem:** after a while of use, old tournaments start piling up and it becomes increasingly annoying to find the active ones (both in the sidebar switcher and the frontend). 

**Proposed solution:** An "archive this tournament" switch in the tournament options that:

1. moves the tournament to a collapsed-by-default "archived" section in the sidebar

![image](https://user-images.githubusercontent.com/3891092/146636908-28a68883-f7af-49ee-9268-2fe0a75aaec2.png)

2. moves their links in the frontend below the non-archived ones

![image](https://user-images.githubusercontent.com/3891092/146636992-fcbf30eb-f3a5-43ae-9ac2-aebb3ceb7a5e.png)

*(screenshots are just mockups)*

Having this flag available would also enable potential performance optimizations if they become necessary (like extending the frontend cache time since archived tournaments could be treated as read-only).

Thoughts?

**Comment by czlee** [comment]
Ah, interesting. So, I think we kind of have this. But maybe it's not as prominent as it could be, or maybe it's poorly named.

Tournaments currently have a field called "active", and it's true by default. If you set it to false, then it should disappear from the main list on the home page, and in #1384, @tienne-B added a page for inactive tournaments. Is this basically the functionality you're suggesting? Maybe we should rename "inactive" to "archived" or something?

Someone might correct me, but from memory to set a tournament as inactive you need to do it via Edit Database—if this is frequently used, potentially it could be brought out to an admin area page in the Configuration section or something.

Thanks so much for these suggestions and willingness to contribute!

**Comment by franga2000** [comment]
Oh wow, that's exactly what I had in mind! It is indeed only available from Edit Database, but that seems pretty reasonable to me. 

I'm tempted to support renaming it to "archived" in the frontend, as while "inactive" is entirely correct, at least to me it doesn't convey the same meaning of "this tournament was active, but is now finished". Then again, I can see some use-cases where the "inactive" tag wouldn't be permanent, in which case "archived" would actually be incorrect (like if a split-timezone tournament wanted to indicate which division is currently active). 

So...looks good as it is, I might still send a PR for the admin dropdown one day if it doesn't break anything, and I'll look around a bit harder next time before opening an issue :sweat_smile:

**Comment by czlee** [comment]
Yeah, tbh I think the original intention was probably closer to "archive" (@philipbelesky?), so we should probably rename it 😃

If we want to avoid a schema change/database migration, just changing the UI is fine. But I think it's also fine to rename `active` to `archived` (and to flip the values, so this would require a data migration to toggle the whole column). I don't really mind either way.

Thanks for raising this, @franga2000! The fact that it was not obvious is very useful feedback.

**Comment by franga2000** [comment]
I'm coming back to this issue after #2053 was merged, because our Tabbycat front page is now quite crowded again due to all the inactive tournaments showing back up. 

Would you be open to a accepting a PR that adds a site-wide setting (under "Global settings") to toggle between the current and old behaviour (where inactive tournaments were not on the front page and required a click to get to)?

Something like **"Show inactive tournaments on home page"** with a description like *"Inactive tournaments will be shown directly below active ones on the public page, instead of being on separate "archive" page"*. I'd turn it on by default, so nothing changes once it's merged, but still gives the option to those of us that prefer the old design.

**Comment by philipbelesky** [comment]
@franga2000 how would you feel about this:

- Homepage always displays all active tournaments
- Homepage always displays first `X` (6?) archived tournaments. If additional archived tournaments exist, create a link to a page that shows the full list
